.github/workflows/pull_request_fc.yml

```diff
@@ -4,13 +4,18 @@ on: [pull_request]
 
 jobs:
   build:
-    runs-on: ubuntu-latest
+    runs-on: [self-hosted, "${{ matrix.arch }}" ]
+    strategy:
+      matrix:
+        arch: [x86_64, aarch64]
+      fail-fast: false
```

Comment:
Do we need fail-fast to be false here? Unless I'm missing something we can just fail whenever a step fails

Reply:
fail-fast refers to job statuses, not steps. Its declaration here means that the two (since matrix has two values) jobs running in parallel will not stop running should one of them fails. They are independent. If a problem occurs in x86, aarch64 doesn't have to stop running.

Reply:
Oh ok, I didn't know that. In that case it's fine, however I'm confused by line 53:

`echo ::set-output name=build_exit_code::$?`

and related line 57:

` if: ${{ (steps.build-vmlinux.outputs.build_exit_code == 0 && steps.build-virtio.outputs.build_exit_code == 0) }}`

Reply:
these lines are unnecessary but not harmful. I wrote them because I wasn't sure how step failure is defined and I used this check preemptively in case one of the artifacts is not built successfully but the Actions software does not register it as a failure for the entire step. I haven't been able to find an answer whether a return value different than 0 for a command is always equal to step failure. I can remove them however

Reply:
I think it does. If whatever is in the `run` section returns something other than zero the step fails, so yes, I would remove these lines. There's one in the step that builds the kernel as well which I missed earlier.